### DISCUSSION TITLE
Upgrading from 11 to 12.

### DISCUSSION THREAD

**Discussion by Dropbear67** [blakeblackshear/frigate#6065] [Ask A Question]
Looking for a fairly easy to follow upgrade guide from 11 to 12. What needs to be changed if anything in the config file, how do I get docker to pull down the new image etc. Bit of a noob but I did get 11 working ok.

**Comment by groenmarsmannetje** [comment]
I totally agree. This new release should have come with a proper migration guide as it seems that quite a lot has changed. I am afraid that I will loose database and configuration file when I do a new pull in Portainer. Maybe a fresh deploy might even be a better option. I really do not have a clue what is the best approach.

**Reply by NickM-27** [reply]
This release has a migration guide for go2rtc. But that's not required for frigate to startup. ALL of the potential issues are listed in breaking changes, besides those your same config will startup as expected.

**Comment by Karmak23** [comment]
Hi,
I’ve got 10 cameras and a fairly complex setup, all crafted by hand (Frigate via docker-compose + HA docker supervised). I migrated from 0.11 to 0.12 yesterday night and:
- i did not backup my database and all went OK (don’t do this in real life situations).
- the 0.11 configuration file worked out of the box (RTMP is deprecated but still supported, you don't get go2rtc but it’s not mandatory anyway…)
- even the HA frigate integration and cards worked correctly.

My `docker-compose.yml` file missed only the new `ports` (1934, 8554, 8555), `shm` parameters where already correct (in 0.10 i had set up twice the needs given by the calculator).

I adapted the Frigate `config.yml` after migration, but only to ADD new features like go2rtc to get the immediate live streams, restream feature, and simplifications like ffmpeg preset aliases… AFAIR, I did:

- create the go2rtc section from scratch, reusing the stream path of my cameras already present in the file. Excerpt for 2 cameras:
```yaml
go2rtc:
  webrtc:
    candidates:
      - 192.168.1.11:8555
      - stun:8555
  streams:
    AbriBois:
      - rtsp://user1:pass1@cam1.mydomain.lan/stream1
    Jardin:
      - rtsp://user2:pass2@cam2.mydomain.lan/stream1

```
- simplified the paths of the camera streams to reuse the ones from go2rtc section. Excerpt for one camera:
```yaml
        - path: rtsp://127.0.0.1:8554/AbriBois
          input_args: preset-rtsp-restream
```
- added `restream: True` in the `birdseye` section

NOTICE: in camera sections there is a `path:` keyword ; in `go2rtc` section it must **not** be there (you put the stream path directly, else you get a cryptic error leading you to an already open issue on `Frigate` github telling you to remove `path:` in `go2rtc` section ;-) ). 

I think that besides the database, you cannot "loose" anything, because your data is outside the docker container. Just stop the container, copy the whole data directory to have a backup, and `docker-compose` `rm`/`pull`/`build`/`up --build -d`. That’s what I did.

**Reply by Dropbear67** [reply]
I'd just like to say thank you. This helped a lot and my upgrade and conversion to go2rtc has finished successfully.

**Comment by NickM-27** [comment]
Besides the items in the breaking changes section, your current config should startup just the same
